sphere-models/src/it/java/io/sphere/sdk/products/messages/ProductPublishedMessageTest.java

```diff
@@ -0,0 +1,32 @@
+package io.sphere.sdk.products.messages;
+
+import io.sphere.sdk.messages.queries.MessageQuery;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.ProductFixtures;
+import io.sphere.sdk.products.commands.ProductUpdateCommand;
+import io.sphere.sdk.products.commands.updateactions.Publish;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.test.IntegrationTest;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProductPublishedMessageTest extends IntegrationTest {
+    @Test
+    public void message() {
+        ProductFixtures.withUpdateableProduct(client(), product -> {
+            assertThat(product.getMasterData().isPublished()).isFalse();
+            final Product publishedProduct = execute(ProductUpdateCommand.of(product, Publish.of()));
+            final PagedQueryResult<ProductPublishedMessage> queryResult = execute(MessageQuery.of()
+                    .withSort(m -> m.createdAt().sort().desc())
+                    .withExpansionPaths(m -> m.resource())
+                    .withLimit(1)
+                    .forMessageType(ProductPublishedMessage.MESSAGE_HINT));
```

Comment:
Since here you are naming the method `forMessageType`, it doesn't match with the constant `MESSAGE_HINT`. I think it would be better to call both either MessageType or MessageHint.